### DISCUSSION TITLE
DAGs disappearing from UI, but exist in DB with active flag = 0 on tag and owner change

### DISCUSSION THREAD

**Discussion by khannavivek** [apache/airflow#39779] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.9.0

### What happened?

DAGs disappear from UI but exist in DB with active flag = 0 on tag and owner change.

Tried hit and trial method to identify the issue but was not able to identify what exactly was causing this, sometimes owner's name in all caps disappears the dag or vice versa.

### How to reproduce

Not sure how to reproduce, because happing very randomly

### Operating System

Ubuntu 22.04.4 LTS

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
Please provide more details about your environment and a small reproducible example, which will able to be reproduce on someone other environment.
